app/models/miq_server/worker_management/monitor/start.rb

```diff
@@ -5,12 +5,23 @@ def wait_for_started_poll
     (@worker_monitor_settings[:wait_for_started_poll] || 10.seconds).to_i_with_method
   end
 
+  def update_non_rails_worker_status(starting)
+    miq_services = sync_from_system
+    starting.each do |worker|
+      next if worker.class.rails_worker?
+
+      systemd_worker = miq_services.find { |w| w[:name].include?(worker[:guid]) }
+      worker.update!(:status => "started") if systemd_worker[:load_state] == "loaded" && systemd_worker[:active_state] == "active" && systemd_worker[:sub_state] == "running"
+    end
+  end
+
   def wait_for_started_workers
     last_which                    = nil
     entered_wait_for_started_loop = Time.now.utc
     wait_for_started_timeout      = @worker_monitor_settings[:wait_for_started_timeout] || 10.minutes
     loop do
       starting = MiqWorker.find_starting.find_all { |w| MiqWorkerType.worker_class_names.include?(w.class.name) }
+      update_non_rails_worker_status(starting)
       if starting.empty?
```

Comment:
I think we should "sync" the worker status before doing the `MiqWorker.find_starting.find_all` query, since the way this is now a worker that is "starting" in the database but has `SdNotify.ready` already will be in the `starting` result, marked as started in the database, then logged as not started for the rest of this block.

Reply:
Or we need to drop the worker record from the `starting` set